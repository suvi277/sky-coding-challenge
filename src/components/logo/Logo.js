import React from 'react';

const Logo = () => {
	return (
		<a href="/" className="navbar-brand">
			<img src="https://www.sky.com/assets/masthead/images/sky-logo.png" alt="logo" width="50" />
		</a>
	);
};
export default Logo;
