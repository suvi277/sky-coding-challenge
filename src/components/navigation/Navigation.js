import React from 'react';
import { ROUTES } from 'routes/routeConfig';
import { NavLink } from 'react-router-dom';
import './Navigation.scss';

const Navigation = ({ routes }) => {
	const getNavList = (route, i) => {
		let navItems = '';

		if (route.key !== 'notFound') {
			navItems = (
				<li key={i} className="nav-item">
					<NavLink className="nav-link p-3" to={route.path} exact={route.exact}>
						{route.label}
					</NavLink>
				</li>
			);
		}
		return navItems;
	};
	return (
		<ul className="nav justify-content-end">
			{ROUTES.map((route, i) => {
				return getNavList(route, i);
			})}
		</ul>
	);
};

export default Navigation;
