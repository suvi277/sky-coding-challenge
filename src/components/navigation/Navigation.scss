@import "../../_shared/styles/variable";

.nav-link {
	&.active {
		background-color: $primary;
        color: $light;
    }
    
    &:not(.active):hover {
        background-color: $secondary;
		color: $light;
    }
}
