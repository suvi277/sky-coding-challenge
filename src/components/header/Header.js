import React from 'react';
import Navigation from 'components/navigation/Navigation';
import Logo from 'components/logo/Logo';

const Header = ({ routes }) => {
	return (
		<header className="sticky-top navbar bg-dark p-0">
			<div className="container">
				<Logo />
				<Navigation routes={routes} />
			</div>
		</header>
	);
};
export default Header;
