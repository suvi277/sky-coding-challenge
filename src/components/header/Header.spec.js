import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import Navigation from 'components/navigation/Navigation';
import Logo from 'components/logo/Logo';

describe('Navigation Component', () => {
	const mockRouteConfig = [
		{
			path: '/',
			key: 'home',
			label: 'Home',
			exact: true,
			component: () => {}
		},
		{
			path: '/content',
			key: 'streams',
			label: 'Streams',
			component: () => {}
		}
	];
	const wrapper = shallow(<Header routes={mockRouteConfig} />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should exist Logo Component', () => {
		expect(wrapper.exists(Logo)).toBeTruthy();
	});
	it('should exist Navigation Component', () => {
		expect(wrapper.exists(Navigation)).toBeTruthy();
	});
	it('should render Navigation Component with routes props', () => {
		const navigation = wrapper.find(Navigation);
		expect(navigation.props().routes).toBe(mockRouteConfig);
	});
});
