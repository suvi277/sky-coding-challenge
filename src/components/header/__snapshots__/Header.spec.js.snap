// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders as expected 1`] = `
<header
  className="sticky-top navbar bg-dark p-0"
>
  <div
    className="container"
  >
    <Logo />
    <Navigation
      routes={
        Array [
          Object {
            "component": [Function],
            "exact": true,
            "key": "home",
            "label": "Home",
            "path": "/",
          },
          Object {
            "component": [Function],
            "key": "streams",
            "label": "Streams",
            "path": "/content",
          },
        ]
      }
    />
  </div>
</header>
`;
